### ISSUE TITLE
Code folding fails with $$ sign and {} in the same line

### ISSUE THREAD

**Issue by Jacobi-Leo** [neilsustc/vscode-markdown#194]
Whenever this extension is enabled, the folding behavior will fail after one special with both $$ sign and {}.

```
Test File
=========

Section one
-----------

$$ a^2 + b^2 = c^2 $$ {#eq:triangle}

Section two
-----------

Something...
```
fails to fold with 'Section two' folded under 'Section one' instead being parallel. Once the {} is removed, the behavior is correct.

My VS Code version is 

> 1.23.1
>d0182c3417d225529c6d5ad24b7572815d0de9ac
>x64

under Linux.

No error message.

**Comment by Jacobi-Leo** [comment]
A piece of additional information: the outline generated in EXPLORER bar is corrected, only the folding fails.

**Comment by neilsustc** [comment]
Good catch. However, the folding is provided by VSCode built-in Markdown extension. You can report this to repository Microsoft/vscode.

**Comment by neilsustc** [comment]
I find that it is also not an issue of vscode... I'll provide more information there.

**Comment by neilsustc** [comment]
It seems the math part in `$$ a^2 + b^2 = c^2 $$ {#eq:triangle}` will not be treated as math block. I suggest not using this kind of syntax.
